src/character.cpp

```diff
@@ -1491,6 +1497,14 @@ bool Character::can_run() const
     return get_stamina() > 0 && !has_effect( effect_winded ) && get_working_leg_count() >= 2;
 }
 
+bool Character::is_bp_armored( const bodypart_id &bp ) const
+{
+    return ( bp->token == bp_torso && has_bionic( bio_armor_torso ) ) || \
```

Comment:
Same here, don't hardcode CBMs. Instead you should either make it a json property or check armor bonus form bionic on the tested bodypart